modules/mod_ls.c

```diff
@@ -3368,6 +3368,9 @@ MODRET ls_nlst(cmd_rec *cmd) {
       if (pr_data_open(NULL, "file list", PR_NETIO_IO_WR, 0) < 0) {
         int xerrno = errno;
 
+        pr_response_add_err(R_425, "%s: %s", (char *) cmd->argv[0],
```

Comment:
This addition makes sense -- but why not keep the existing response codes as they are?  My understanding is that the discrepancy in response codes is what's confusing your legacy FTP client.